Sage/carbon
# feat(focus-trap): custom selector for focusable elements

See details of the feature in the following issue: https://github.com/Sage/carbon/issues/5555

### Checklist



- [N/A] Screenshots are included in the PR if useful
- [N/A] All themes are supported if required
- [N/A] Cypress automation tests added or updated if required
- [N/A] Storybook added or updated if required
- [N/A] Translations added or updated (including creating or amending translation keys table in storybook) if required

#### QA


### Additional context



### Testing instructions

1. Create a dialog with a few buttons
2. Add a few focusable elements with special selector
3. Pass the selector that matches only some of your focusable elements to the dialog using the new `focusableSelectors` prop
4. The focus should move to only the fields that the selector passed in `focusableSelectors` matches

I'll add a sandbox example once it gets build by codesandbox.

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
